src/libjson/json.c

```diff
@@ -0,0 +1,1217 @@
+** notice and this permission notice shall be included in all copies or
+** substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+** OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+** THE SOFTWARE.
+**
+**==============================================================================
+*/
+
+#include "json.h"
+#include "raise.h"
+
+/*
+**==============================================================================
+**
+** libc compatibility:
+**
+**==============================================================================
+*/
+
+#ifndef NULL
+#define NULL ((void*)0)
+#endif
+
+typedef unsigned long size_t;
+typedef unsigned int uint32_t;
+typedef unsigned long uint64_t;
+typedef long ptrdiff_t;
+
+static int _tolower(int c)
+{
+    return (c >= 'A' && c <= 'Z') ? (c - ' ') : c;
+}
+
+static int _isdigit(int c)
+{
+    return (c >= '0' && c <= '9');
+}
+
+static int _isspace(int c)
+{
+    return c == ' ' || c == '\f' || c == '\n' ||
+        c == '\r' || c == '\t' || c == '\v';
+}
+
+static void* _memset(void* s, int c, size_t n)
+{
+    unsigned char* p = (unsigned char*)s;
+
+    while (n--)
+        *p++ = c;
+
+    return s;
+}
+
+static void* _memcpy(void* dest, const void* src, size_t n)
+{
+    unsigned char* p = (unsigned char*)dest;
+    unsigned char* q = (unsigned char*)src;
+
+    while (n--)
+        *p++ = *q++;
+
+    return dest;
+}
+
+static int _memcmp(const void* s1, const void* s2, size_t n)
+{
+    unsigned char* p = (unsigned char*)s1;
+    unsigned char* q = (unsigned char*)s2;
+
+    while (n--)
+    {
+        if (*p < *q)
+            return -1;
+        else if (*p > *q)
+            return 1;
+
+        p++;
+        q++;
+    }
+
+    return 0;
+}
+
+static void* _memmove(void* dest_, const void* src_, size_t n)
+{
+    char *dest = (char*)dest_;
+    const char *src = (const char*)src_;
+
+    if (dest != src && n > 0)
+    {
+        if (dest <= src)
+        {
+            _memcpy(dest, src, n);
+        }
+        else
+        {
+            for (src += n, dest += n; n--; dest--, src--)
+                dest[-1] = src[-1];
+        }
+    }
+
+    return dest;
+}
+
+static size_t _strlen(const char* s)
+{
+    const char* p = (const char*)s;
+
+    while (*p++)
+        ;
+
+    return p - s;
+}
+
+static int _strcmp(const char* s1, const char* s2)
+{
+    while ((*s1 && *s2) && (*s1 == *s2))
+    {
+        s1++;
+        s2++;
+    }
+
+    return *s1 - *s2;
+}
+
+static size_t _strlcpy(char* dest, const char* src, size_t size)
```

Comment:
Thanks for re-implementing these functions to cut the dependencies, that makes everything a bit easier!

Reply:
Thank you. I learned my lesson this time. Hopefully this is 100% standalone now. :)